### PR TITLE
Replace random function to make CI reproducible

### PR DESCRIPTION
**Pull request by Koren-Brand**
## Describe the changes

This PR changes random generation to be based on rand such that bugs could be effectively reproduced.
Ctest (CI) Now outputs test log on test failure

## Linked Issues

Resolves #

### CODE REVIEW

File: icicle/include/icicle/fields/field.h [hunk 1]
```diff
@@ -691,12 +691,9 @@ class Field
 
   static HOST_INLINE Field rand_host()
   {
-    std::random_device rd;
-    std::mt19937_64 generator(rd());
-    std::uniform_int_distribution<unsigned> distribution;
     Field value{};
     for (unsigned i = 0; i < TLC; i++)
-      value.limbs_storage.limbs[i] = distribution(generator);
+      value.limbs_storage.limbs[i] = rand();
```
Comment (mickeyasa) [inline icicle/include/icicle/fields/field.h:696]: the uniform distribution does nothing?

**Review by mickeyasa — APPROVED** [review]
good initiative

File: icicle/include/icicle/fields/field.h (hunk 1 above)
Comment (yshekel) [inline icicle/include/icicle/fields/field.h:696]: maybe assert that limbs is 4 bytes here because rand() returns int

**Review by yshekel — APPROVED** [review]
Approved but please see the comment

File: icicle/include/icicle/fields/field.h (hunk 1 above)
Comment (Koren-Brand) [reply icicle/include/icicle/fields/field.h:696]: The implementation of limbs_storage in field is uint32 so it is an exact fit

Comment (Koren-Brand) [reply icicle/include/icicle/fields/field.h:696]: Apparently, rand has a maximal value of 2^15 - 1 so another solution might be better

File: icicle/include/icicle/fields/field.h [hunk 2]
```diff
@@ -689,14 +689,18 @@ class Field
     return rv;
   }
 
+  inline static std::mt19937 field_rand_generator = std::mt19937{};
```
Comment (yshekel) [inline icicle/include/icicle/fields/field.h:692]: maybe better if you moved all of this to utils and use it here from utils.

Comment (Koren-Brand) [reply icicle/include/icicle/fields/field.h:692]: that's a good idea

File: icicle/include/icicle/fields/field.h [hunk 3]
```diff
@@ -24,6 +24,7 @@
 #endif // __CUDACC__
 
 #include "icicle/errors.h"
+#include "icicle/utils/rand_gen.cpp"
```
Comment (yshekel) [inline icicle/include/icicle/fields/field.h:27]: you need to rename it to rand_gen.h

File: icicle/include/icicle/utils/rand_gen.cpp [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (yshekel) [inline icicle/include/icicle/utils/rand_gen.cpp:1]: it's not a c++ file right? rename to .h

Comment (Koren-Brand) [reply icicle/include/icicle/utils/rand_gen.cpp]: changed

**Pull request merged**